table/table_iterator.h

```diff
@@ -48,22 +49,65 @@ class TableIterator : public Iterator {
   bool Valid() const override { return iter_->Valid(); }
   void SeekToFirst() override { return iter_->SeekToFirst(); }
   void SeekToLast() override { return iter_->SeekToLast(); }
-  void Seek(const Slice& target) override { return iter_->Seek(target); }
+  void Seek(const Slice& target) override {
+      std::string seek_key_buf;
+      ROCKSDB_NAMESPACE::Status s = ROCKSDB_NAMESPACE::GetInternalKeyForSeek(target, options_->comparator,
+                                                                             &seek_key_buf);
+      assert(s.ok());
+      if (!s.ok()) {
+        status_ = s;
+      }
+      iter_->Seek(seek_key_buf);
+  }
   void SeekForPrev(const Slice& target) override {
-    return iter_->SeekForPrev(target);
+      std::string seek_key_buf;
```

Comment:
@ajkr @pdillinger Can you comment on this change in the internals for RocksDB please?